### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by WalkerJim1**
Created solution to Improved UNet ISIC skin lesion segmentation problem, based on paper: https://arxiv.org/pdf/1802.10508v1.pdf

README contains details on usage and model with results

Files:
dataset.py      handles loading in ISIC dataset
modules.py    creates Improved UNet structure
train.py           trains the model from scratch on training dataset
predict.py       visualises the data, makes predictions

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
This is an initial inspection, no action is required at this point.

- Wrong files path. All student constructed document should be packed in one document under the ‘recognition’ document.
- Algorithm solves the problem appropriately. However, no Average Dice Coefficient in testing dataset. Predicted Mask output is not binary one, so looks a little bit blur. Other parts are ok.
- Correctly using PyTorch to construct the Improved U-Net models and functions.
- Good coding design. Good handwriting code.
- Sufficient docstring and comment in the code.
- Easy difficulty problem, no higher difficulty marks.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, but no results on test set -1
Difficulty: Easy -10
## Commit Log
Meaningful commit messages sometimes, could be better -1
Progressive commits used
## Documentation
ReadMe acceptable, missing usage
Model/technical explanation lacking -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required move readme and restore repo README. -2
Request Description good

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.
